alexherlanda/db8-events
# Internationalization Structure 🌎

## Abstract 
Adds the ability to the front to load strings dynamically based in a selected language variable and resource files for each language.

## Purpose
Letting the user select their preferred language to interact with the platform to increase accessibility and increase comfort in the UX

## Summary 
- Adds new dependencies i18next and react-i18next 
- Adds folder structure to add translations
- Adds basic configuration of i18next and react-i18next 
- Other minor fixes for the front

## How to test
- Run `yarn install` to update dependencies
- Run `yarn start`
- Edit line 19 on client/src/i18n/index.js and change the parameter to en, pt, or 'es'
- Check that the main button changes based on this parameter

## An image tells more than 1000 words
![image](https://user-images.githubusercontent.com/43116659/86989956-2a118080-c161-11ea-96cc-128553a4b40e.png)
![image](https://user-images.githubusercontent.com/43116659/86989967-3269bb80-c161-11ea-9431-390b37a4dce3.png)


## Next steps 
- Add the UI for selecting the language
- Add all the translations
